### ISSUE TITLE
macOS Crash Rika Bravo

### ISSUE THREAD

**Issue by dillontaylor2** [HarbourMasters/2ship2harkinian#727]
### Description of the crash:
Simply opening the application.

### Can you reproduce? If so please list the steps:
- [ ] No, I cannot reproduce this crash
- [✅] Yes, here are the steps to reproduce the crash:

### Logs:

-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               2s2h-macos [19369]
Path:                  /private/var/folders/*/2s2h.app/Contents/Resources/2s2h-macos
Identifier:            2s2h-macos
Version:               ???
Code Type:             ARM-64 (Native)
Parent Process:        Exited process [19367]
User ID:               501

Date/Time:             2024-06-28 11:53:35.7376 -0400
OS Version:            macOS 14.5 (23F79)
Report Version:        12
Anonymous UUID:        1890C434-863B-0C54-DC65-3A55A2F706BB

Sleep/Wake UUID:       7C059DA7-F320-4F6A-AC09-16DF44844847

Time Awake Since Boot: 110000 seconds
Time Since Wake:       2052 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Termination Reason:    Namespace SIGNAL, Code 6 Abort trap: 6
Terminating Process:   2s2h-macos [19369]

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x182a52a60 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x182a8ac20 pthread_kill + 288
2   libsystem_c.dylib             	       0x182997a30 abort + 180
3   libsystem_c.dylib             	       0x182996d20 __assert_rtn + 284
4   Metal                         	       0x18cf49194 MTLReportFailure.cold.1 + 48
5   Metal                         	       0x18cf25db0 MTLReportFailure + 464
6   Metal                         	       0x18cf1bd58 _MTLMessageContextEnd + 876
7   Metal                         	       0x18cf23018 validateWithDevice(id<MTLDevice>, MTLRenderPipelineDescriptorPrivate const&) + 556
8   Metal                         	       0x18cf22d68 -[MTLRenderPipelineDescriptorInternal validateWithDevice:error:] + 96
9   AGXMetalG15X_B0               	       0x1dd1a3180 AGX::UserCommonShaderFactory<AGX::G15::Encoders, AGX::G15::Classes, AGX::G15::ObjClasses>::createRenderPipeline(MTLRenderPipelineDescriptor*, AGXG15XFamilyDevice*, unsigned long, MTLRenderPipelineReflection**, NSError**, void (id<MTLRenderPipelineState>, MTLRenderPipelineReflection*, NSError*) block_pointer) + 140
10  AGXMetalG15X_B0               	       0x1dd1b17d8 -[AGXG15XFamilyDevice newRenderPipelineStateWithDescriptor:error:] + 116
11  2s2h-macos                    	       0x1030fbb14 gfx_metal_create_and_load_new_shader(unsigned long long, unsigned int) + 1172
12  2s2h-macos                    	       0x10310fa7c gfx_sp_tri1(unsigned char, unsigned char, unsigned char, bool) + 5980
13  2s2h-macos                    	       0x1031182d0 gfx_draw_rectangle(int, int, int, int) + 380
14  2s2h-macos                    	       0x1031125e8 gfx_dp_texture_rectangle(int, int, int, int, unsigned char, short, short, short, short, bool) + 240
15  2s2h-macos                    	       0x1031124e8 gfx_tex_rect_and_flip_handler_rdp(F3DGfx**) + 112
16  2s2h-macos                    	       0x10311429c gfx_run(Gfx*, std::__1::unordered_map<MtxS*, MtxF, std::__1::hash<MtxS*>, std::__1::equal_to<MtxS*>, std::__1::allocator<std::__1::pair<MtxS* const, MtxF>>> const&) + 1136
17  2s2h-macos                    	       0x102b65264 Graph_ProcessGfxCommands + 932
18  2s2h-macos                    	       0x102c5ded4 Graph_ExecuteAndDraw + 820
19  2s2h-macos                    	       0x102c5dfdc Graph_Update + 80
20  2s2h-macos                    	       0x102c5e224 RunFrame + 460
21  2s2h-macos                    	       0x102c5e250 Graph_ThreadEntry + 20
22  2s2h-macos                    	       0x102c5ed48 main + 272
23  dyld                          	       0x1827020e0 start + 2360

Thread 1:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1032aa010 spdlog::details::mpmc_blocking_queue<spdlog::details::async_msg>::dequeue(spdlog::details::async_msg&) + 76
4   2s2h-macos                    	       0x1032a9ec8 spdlog::details::thread_pool::process_next_msg_() + 140
5   2s2h-macos                    	       0x1032aa360 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, spdlog::details::thread_pool::thread_pool(unsigned long, unsigned long, std::__1::function<void ()>, std::__1::function<void ()>)::$_4>>(void*) + 76
6   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
7   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 2:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1030dfb44 BS::thread_pool::worker(unsigned int, std::__1::function<void ()> const&) + 328
4   2s2h-macos                    	       0x1030e0540 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (BS::thread_pool::*)(unsigned int, std::__1::function<void ()> const&), BS::thread_pool*, unsigned int, std::__1::function<void ()>>>(void*) + 84
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 3:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1030dfb44 BS::thread_pool::worker(unsigned int, std::__1::function<void ()> const&) + 328
4   2s2h-macos                    	       0x1030e0540 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (BS::thread_pool::*)(unsigned int, std::__1::function<void ()> const&), BS::thread_pool*, unsigned int, std::__1::function<void ()>>>(void*) + 84
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 4:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1030dfb44 BS::thread_pool::worker(unsigned int, std::__1::function<void ()> const&) + 328
4   2s2h-macos                    	       0x1030e0540 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (BS::thread_pool::*)(unsigned int, std::__1::function<void ()> const&), BS::thread_pool*, unsigned int, std::__1::function<void ()>>>(void*) + 84
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 5:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1030dfb44 BS::thread_pool::worker(unsigned int, std::__1::function<void ()> const&) + 328
4   2s2h-macos                    	       0x1030e0540 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (BS::thread_pool::*)(unsigned int, std::__1::function<void ()> const&), BS::thread_pool*, unsigned int, std::__1::function<void ()>>>(void*) + 84
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 6:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1030dfb44 BS::thread_pool::worker(unsigned int, std::__1::function<void ()> const&) + 328
4   2s2h-macos                    	       0x1030e0540 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (BS::thread_pool::*)(unsigned int, std::__1::function<void ()> const&), BS::thread_pool*, unsigned int, std::__1::function<void ()>>>(void*) + 84
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 7:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1030dfb44 BS::thread_pool::worker(unsigned int, std::__1::function<void ()> const&) + 328
4   2s2h-macos                    	       0x1030e0540 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (BS::thread_pool::*)(unsigned int, std::__1::function<void ()> const&), BS::thread_pool*, unsigned int, std::__1::function<void ()>>>(void*) + 84
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 8:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1030dfb44 BS::thread_pool::worker(unsigned int, std::__1::function<void ()> const&) + 328
4   2s2h-macos                    	       0x1030e0540 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (BS::thread_pool::*)(unsigned int, std::__1::function<void ()> const&), BS::thread_pool*, unsigned int, std::__1::function<void ()>>>(void*) + 84
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 9:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x1030dfb44 BS::thread_pool::worker(unsigned int, std::__1::function<void ()> const&) + 328
4   2s2h-macos                    	       0x1030e0540 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (BS::thread_pool::*)(unsigned int, std::__1::function<void ()> const&), BS::thread_pool*, unsigned int, std::__1::function<void ()>>>(void*) + 84
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 10:
0   libsystem_pthread.dylib       	       0x182a85d20 start_wqthread + 0

Thread 11:
0   libsystem_pthread.dylib       	       0x182a85d20 start_wqthread + 0

Thread 12:
0   libsystem_pthread.dylib       	       0x182a85d20 start_wqthread + 0

Thread 13:: caulk.messenger.shared:17
0   libsystem_kernel.dylib        	       0x182a4a170 semaphore_wait_trap + 8
1   caulk                         	       0x18d003624 caulk::semaphore::timed_wait(double) + 212
2   caulk                         	       0x18d0034d8 caulk::concurrent::details::worker_thread::run() + 36
3   caulk                         	       0x18d0031d8 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*>>>(void*) + 96
4   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
5   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 14:: caulk.messenger.shared:high
0   libsystem_kernel.dylib        	       0x182a4a170 semaphore_wait_trap + 8
1   caulk                         	       0x18d003624 caulk::semaphore::timed_wait(double) + 212
2   caulk                         	       0x18d0034d8 caulk::concurrent::details::worker_thread::run() + 36
3   caulk                         	       0x18d0031d8 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*>>>(void*) + 96
4   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
5   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 15:: AudioQueue thread
0   libsystem_kernel.dylib        	       0x182a4a1f4 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x182a5cb24 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x182a52e34 mach_msg_overwrite + 476
3   libsystem_kernel.dylib        	       0x182a4a578 mach_msg + 24
4   CoreFoundation                	       0x182b6a680 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x182b68f44 __CFRunLoopRun + 1208
6   CoreFoundation                	       0x182b68434 CFRunLoopRunSpecific + 608
7   libSDL2-2.0.0.dylib           	       0x104905d94 audioqueue_thread + 1136
8   libSDL2-2.0.0.dylib           	       0x1048988a4 SDL_RunThread + 48
9   libSDL2-2.0.0.dylib           	       0x1048f15fc RunThread + 12
10  libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
11  libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 16:: caulk::deferred_logger
0   libsystem_kernel.dylib        	       0x182a4a170 semaphore_wait_trap + 8
1   caulk                         	       0x18d003624 caulk::semaphore::timed_wait(double) + 212
2   caulk                         	       0x18d0034d8 caulk::concurrent::details::worker_thread::run() + 36
3   caulk                         	       0x18d0031d8 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*>>>(void*) + 96
4   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
5   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 17:: AQConverterThread
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libAudioToolboxUtility.dylib  	       0x191af3ce4 CADeprecated::CAGuard::Wait() + 76
3   AudioToolbox                  	       0x192dcaf78 AQConverterManager::AQConverterThread::ConverterThreadEntry(void*) + 572
4   libAudioToolboxUtility.dylib  	       0x191ad0254 CADeprecated::CAPThread::Entry(CADeprecated::CAPThread*) + 92
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 18:: com.apple.audio.IOThread.client
0   libsystem_kernel.dylib        	       0x182a4a1f4 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x182a5cb24 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x182a52e34 mach_msg_overwrite + 476
3   libsystem_kernel.dylib        	       0x182a4a578 mach_msg + 24
4   CoreAudio                     	       0x1854279f0 HALB_MachPort::SendSimpleMessageWithSimpleReply(unsigned int, unsigned int, int, int&, bool, unsigned int) + 96
5   CoreAudio                     	       0x1852fbfe0 HALC_ProxyIOContext::IOWorkLoop() + 4272
6   CoreAudio                     	       0x1852fa7f8 invocation function for block in HALC_ProxyIOContext::HALC_ProxyIOContext(unsigned int, unsigned int) + 108
7   CoreAudio                     	       0x1854805c4 HALC_IOThread::Entry(void*) + 88
8   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
9   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 19:
0   libsystem_kernel.dylib        	       0x182a4d9ec __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x182a8b55c _pthread_cond_wait + 1228
2   libc++.1.dylib                	       0x1829b0b14 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3   2s2h-macos                    	       0x102b62b7c OTRAudio_Thread() + 172
4   2s2h-macos                    	       0x102b6d958 void* std::__1::__thread_proxy[abi:v160006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (*)()>>(void*) + 52
5   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 20:
0   libsystem_pthread.dylib       	       0x182a85d20 start_wqthread + 0

Thread 21:
0   libsystem_pthread.dylib       	       0x182a85d20 start_wqthread + 0

Thread 22:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	       0x182a4a1f4 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x182a5cb24 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x182a52e34 mach_msg_overwrite + 476
3   libsystem_kernel.dylib        	       0x182a4a578 mach_msg + 24
4   CoreFoundation                	       0x182b6a680 __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x182b68f44 __CFRunLoopRun + 1208
6   CoreFoundation                	       0x182b68434 CFRunLoopRunSpecific + 608
7   AppKit                        	       0x1864f2188 _NSEventThread + 144
8   libsystem_pthread.dylib       	       0x182a8af94 _pthread_start + 136
9   libsystem_pthread.dylib       	       0x182a85d34 thread_start + 8

Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0x0000000000000000   x3: 0x0000000000000000
    x4: 0x0000000000000073   x5: 0x000000000000002e   x6: 0x0000000000000000   x7: 0x0000000000000000
    x8: 0x33ae66513768cdec   x9: 0x33ae6650dde481ec  x10: 0x000000000000000a  x11: 0x0000000000000000
   x12: 0x0000000000000034  x13: 0x000000011f0521d0  x14: 0x01000001ea8dc589  x15: 0x00000001ea8dc588
   x16: 0x0000000000000148  x17: 0x00000001f4c79928  x18: 0x0000000000000000  x19: 0x0000000000000006
   x20: 0x00000001ea8c4c00  x21: 0x0000000000000103  x22: 0x00000001ea8c4ce0  x23: 0xffffffffffffffff
   x24: 0x00000001e95e6000  x25: 0x00000001f006a000  x26: 0x000000018cf8a8d0  x27: 0x000000016d2a4690
   x28: 0x00006000008ff1a0   fp: 0x000000016d2a4370   lr: 0x0000000182a8ac20
    sp: 0x000000016d2a4350   pc: 0x0000000182a52a60 cpsr: 0x40001000
   far: 0x0000000000000000  esr: 0x56000080  Address size fault

Binary Images:
       0x11ef04000 -        0x11ef0bfff com.apple.GameController.KeyboardAndMouseSupport (*) <136978e3-f554-388b-91fd-2e8e391c92bb> /System/Library/Frameworks/GameController.framework/Versions/A/Resources/KeyboardAndMouseSupport.bundle/Contents/MacOS/KeyboardAndMouseSupport
       0x1371ac000 -        0x137b7ffff com.apple.audio.AudioDSPComponents (1.0) <a060a853-d741-389c-b78b-29ced9c837c7> /System/Library/Components/AudioDSP.component/Contents/MacOS/AudioDSP
       0x11be68000 -        0x11bf9ffff com.apple.audio.units.Components (1.14) <fd93a3bc-5142-33ea-88bd-c2258c5a032d> /System/Library/Components/CoreAudio.component/Contents/MacOS/CoreAudio
       0x104ba4000 -        0x104baffff libobjc-trampolines.dylib (*) <9381bd6d-84a5-3c72-b3b8-88428afa4782> /usr/lib/libobjc-trampolines.dylib
       0x104834000 -        0x10494bfff libSDL2-2.0.0.dylib (*) <f9e212bc-6ae4-3b97-8fd6-58e3318dca4d> /private/var/folders/*/2s2h.app/Contents/Frameworks/libSDL2-2.0.0.dylib
       0x10456c000 -        0x10457bfff libz.1.3.1.dylib (*) <8a6bd02f-4f63-30ae-a288-681ec3508c87> /private/var/folders/*/2s2h.app/Contents/Frameworks/libz.1.3.1.dylib
       0x10452c000 -        0x10453bfff libbz2.1.0.8.dylib (*) <a83fae73-5be6-3f7d-96ab-a2ee3ec964e3> /private/var/folders/*/2s2h.app/Contents/Frameworks/libbz2.1.0.8.dylib
       0x104690000 -        0x1046cbfff libGLEW.2.2.0.dylib (*) <af82adde-851d-3314-a40f-5ccc5bee0d7d> /private/var/folders/*/2s2h.app/Contents/Frameworks/libGLEW.2.2.0.dylib
       0x1045d4000 -        0x1045e7fff libzip.5.5.dylib (*) <beee978d-23ff-35eb-9ff2-6ed829d6723b> /private/var/folders/*/2s2h.app/Contents/Frameworks/libzip.5.5.dylib
       0x10454c000 -        0x104557fff libtinyxml2.10.0.0.dylib (*) <0cbbe0bd-9f1d-3acc-b30b-f21c4487d471> /private/var/folders/*/2s2h.app/Contents/Frameworks/libtinyxml2.10.0.0.dylib
       0x104588000 -        0x1045abfff libpng16.16.dylib (*) <1a580750-10ec-39c5-ae61-98d494fd4643> /private/var/folders/*/2s2h.app/Contents/Frameworks/libpng16.16.dylib
       0x104bf8000 -        0x104cfbfff libiconv.2.dylib (*) <bf47a1df-1c9c-32cf-9c96-922fdb8fd634> /private/var/folders/*/2s2h.app/Contents/Frameworks/libiconv.2.dylib
       0x104718000 -        0x104737fff liblzma.5.dylib (*) <79ab96ba-ccb3-3d5b-aa78-09ad3d5f2170> /private/var/folders/*/2s2h.app/Contents/Frameworks/liblzma.5.dylib
       0x104748000 -        0x1047c3fff libzstd.1.5.6.dylib (*) <ee31e7fc-c1cf-3ddb-b588-350bee714a3b> /private/var/folders/*/2s2h.app/Contents/Frameworks/libzstd.1.5.6.dylib
       0x102b58000 -        0x103413fff 2s2h-macos (*) <51b528be-0e29-38e1-aa66-a8c2479e6ad5> /private/var/folders/*/2s2h.app/Contents/Resources/2s2h-macos
       0x182a49000 -        0x182a83ffb libsystem_kernel.dylib (*) <9b8b53f9-e2b6-36df-98e9-28d8fca732f2> /usr/lib/system/libsystem_kernel.dylib
       0x182a84000 -        0x182a90fff libsystem_pthread.dylib (*) <386b0fc1-7873-3328-8e71-43269fd1b2c7> /usr/lib/system/libsystem_pthread.dylib
       0x182921000 -        0x18299fff7 libsystem_c.dylib (*) <05b44e93-dffc-3bd8-90ab-fd97cb73f171> /usr/lib/system/libsystem_c.dylib
       0x18cdd5000 -        0x18d000fff com.apple.Metal (343.19) <5a9e8280-8897-3f23-b1b1-ea7baf3a02f6> /System/Library/Frameworks/Metal.framework/Versions/A/Metal
       0x1dca7c000 -        0x1dd27bfff com.apple.AGXMetalG15X-B0 (282.14) <da8c84f2-15cc-3f33-947b-8d6a29337e46> /System/Library/Extensions/AGXMetalG15X_B0.bundle/Contents/MacOS/AGXMetalG15X_B0
       0x1826fc000 -        0x182784a17 dyld (*) <37bbc384-0755-31c7-a808-0ed49e44dd8e> /usr/lib/dyld
               0x0 - 0xffffffffffffffff ??? (*) <00000000-0000-0000-0000-000000000000> ???
       0x1829a0000 -        0x182a2cff7 libc++.1.dylib (*) <ec33cd83-7098-3ad6-82c8-bc03ac81e87b> /usr/lib/libc++.1.dylib
       0x18d001000 -        0x18d02afff com.apple.audio.caulk (1.0) <f2da6e2e-b5a0-3ce7-97f5-7d2141a3ec32> /System/Library/PrivateFrameworks/caulk.framework/Versions/A/caulk
       0x182aec000 -        0x182fc4fff com.apple.CoreFoundation (6.9) <84b539d5-22c9-3f8c-84c8-903e9c7b8d29> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
       0x191ac3000 -        0x191afbfff libAudioToolboxUtility.dylib (*) <cf1cffc5-2961-3894-b753-80ac364540d5> /usr/lib/libAudioToolboxUtility.dylib
       0x192da2000 -        0x192f21fff com.apple.audio.toolbox.AudioToolbox (1.14) <b1800481-82d6-39e7-9500-8c14e3d786c9> /System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox
       0x18510e000 -        0x185814fff com.apple.audio.CoreAudio (5.0) <e818958c-f2dc-37df-9896-ad6e9bcd028a> /System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio
       0x18638e000 -        0x1876cafff com.apple.AppKit (6.9) <61f710be-9132-3cc2-883d-066365fba1ad> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

### System Information:
- OS: Sonoma 14.5
- GFX Backend: Metal
- Game Version: 1.0
- Rom Version: 1.0 (U)

**Comment by elliot-wood** [comment]
I'm getting a crash on startup as well, could it be the same thing. I've got the logfile [here](https://github.com/user-attachments/files/16158085/2.Ship.2.Harkinian.log).

The relevant part appears to be:
```
[18:38:06.839] [gfx_metal.cpp:362] [error] Failed to compile shader library, error program_source:62:11: error: assigning to 'float' from incompatible type 'float3' (vector of 3 'float' values)
texel.w = WRAP(texel.w, -0.51, 1.51);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:57: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
program_source:65:9: error: no matching function for call to 'mod'
texel = WRAP(texel, -0.51, 1.51);texel = clamp(texel, 0.0, 1.0);    if (texel.w < 8.0 / 256.0) discard_fragment();
        ^~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:28: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ^~~
program_source:33:8: note: candidate function not viable: no known conversion from 'float4' (vector of 4 'float' values) to 'float3' (vector of 3 'float' values) for 1st argument
float3 mod(float3 a, float3 b) {
       ^
program_source:65:9: error: assigning to 'float4' (vector of 4 'float' values) from incompatible type 'float3' (vector of 3 'float' values)
texel = WRAP(texel, -0.51, 1.51);texel = clamp(texel, 0.0, 1.0);    if (texel.w < 8.0 / 256.0) discard_fragment();
        ^~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:57: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
```

**Comment by inspectredc** [comment]
Hi! Yes that is the same issue and should be fixed by #628 

Going to close this as this is a duplicate of #627
